### PR TITLE
Allow gem to work with Devise 4.2.0

### LINKED ISSUE #34: Devise 4.2.0
There is a new version of Devise [released on 2016-07-01](https://github.com/plataformatec/devise/blob/master/CHANGELOG.md#420---2016-07-01), the [Gemspec of this gem](https://github.com/baschtl/devise-token_authenticatable/blob/99bee7d4ae0bbc694ad993b2fb5d07cef8563047/devise-token_authenticatable.gemspec#L25) tells me I can only use it with Devise < 4.2.

Is there something I can to do get this working on Devise 4.2?

### PR DESCRIPTION
**Pull request by jankeesvw**
This PR fixes #34.

### CODE REVIEW

File: devise-token_authenticatable.gemspec [hunk 1]
```diff
@@ -22,7 +22,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
 
-  spec.add_dependency "devise",                         ">= 4.0.0", "< 4.2.0"
```
Comment (baschtl) [inline devise-token_authenticatable.gemspec]: Could you just include 4.2 with `">= 4.0.0", "< 4.3.0"` as a new 4.3 release might always break everything. ;-)

Could you adapt the table in the [readme](https://github.com/baschtl/devise-token_authenticatable#which-version-to-use-for-which-version-of-devise) accordingly?

Cheers!

**Comment by jankeesvw** [comment]
Hi @baschtl, I think Devise is pretty good with it's SemVer but I understand why you want this. I updated the PR, I think it's ready to merge. ✨

**Comment by baschtl** [comment]
@jankeesvw Thanks!

**Comment by jankeesvw** [comment]
No problem, @baschtl can you also push this version to Rubygems?

**Comment by baschtl** [comment]
Already done! -> https://rubygems.org/gems/devise-token_authenticatable/versions/0.5.2

**Pull request merged**